src/persistence/graph.ts

```diff
@@ -6,15 +6,19 @@ import {IndexedFormula, Statement} from "rdflib";
 export class Graph {
     private persistentStore: any;
     private validationStore: any;
+    private rdfLibStore: any;
     private prefixes: {};
 
     /**
      * Create a new Graph.
      */
     public constructor() {
         let N3 = require("n3");
+        let $rdf = require('rdflib');
+
         this.persistentStore = N3.Store();
```

Comment:
Shout-out to @mweyns and @AlexanderDeSeranno: what's up with these three graphs? `persistentStore` and `validationStore` are n3 graphs and rdflib graphs, respectively. They are kept in sync and manipulated by the persistence logic but AFAICT *they are never actually used.*

Then there's `rdfLibStore`, which is an rdflib graph that is not kept in sync with the other two graphs with quite as much rigor. Though I wager it actually contains the same data most of the time. `rdfLibStore` contains the actual data that is visualized.

Why do we have three copies of the same data? And why do we have an additional dependency on N3 that we don't actually use?

Reply:
Sorry for this somewhat belated reply, but here it is.

The use of rdfLibStore is, I believe, the result of some confusion as to the purpose of the Graph structure. Graph is supposed to adapt/wrap multiple API specific formats so as to facilitate access to these various formats, but mainly to maintain a consistent view of these different representations, i.e. that they all maintain the same view of transformations done on the graph structure, albeit in different formats. The rdfLibStore field, I believe, was introduced  mainly because perhaps it wasn't apparent that $rdf.graph() is essentially sugar (provided by the library) for new IndexedFormula(), but also incidentally circumvents the entire purpose of the Graph structure, which is to maintain that consistent view.

While it isn't strictly true that we don't use N3, I can see where this statement is coming from. N3 is only use for parsing/serialization, i.e. only for persistence. This seems pointless when we could just as well accomplish this with rdflib. As you pointed out, Jonathan, N3 doesn't seem to provide any clear benefits compared to rdflib in the serialization/parsing department, so I will try to refactor this sometime soon.

Regarding this PR, I would recommend removing the rdflibstore and perhaps replacing this.validationStore = new IndexedFormula() with this.validationStore = $rdf.graph(), as this is perhaps clearer than using the underlying type of IndexedFormula. Also, I would advise to refrain from using any setter functions on the underlying stores, as this disrupts the internal consistency of the Graph structure, causing any changes to appear invisible to the persistence layer, which (for now) makes use of a different internal format.